File: web3rpc/rpc-specs/code-samples/javascript/debug/vMTracing/traceBlockByNumber.js
const { JsonRpcProvider } = require("@kaiachain/ethers-ext");
(() => {
  const provider = new JsonRpcProvider("https://public-en-kairos.node.kaia.io");
  const block = 2459;

  provider.debug
    .traceBlockByNumber(block, {}, (err, data, response) => {})
    .then((data) => {
      console.log(data);
    });
})();

File: web3rpc/rpc-specs/code-samples/python/debug/vMTracing/traceBlockByNumber.py
from web3 import Web3
from web3py_ext import extend

host = "https://public-en-kairos.node.kaia.io"

number = 21

w3 = Web3(Web3.HTTPProvider(host))
debug_response = w3.debug.trace_block_by_number(number)

print(debug_response)
